pkg/util/pool/pool.go

```diff
@@ -129,3 +129,34 @@ func (p *BucketedBytes) Put(b *[]byte) {
 		p.usedTotal -= uint64(sz)
 	}
 }
+
+// BatchBytes uses a Bytes pool to hand out byte slices, but they don't need to be
+// individually put back into the pool. Instead, they can be all released at once.
+// Additionally, BatchBytes combines results from the other two
+type BatchBytes struct {
```

Comment:
Given this is shared across multiple chunks reader, I think you have to make it concurrency safe.